From ebd5e15e5b1d0ebdc919c6e2cd982eaaed89d0f8 Mon Sep 17 00:00:00 2001
From: Alex Naidis <alex.naidis@linux.com>
Date: Sat, 20 Aug 2016 01:31:28 +0200
Subject: [PATCH 3/9] skia: boost generation of NEON instructions

* adding the unsafe-math flag increases the code generation ratio for (faster) NEON instructions greatly.
* however this results in a slight loss of precision, which is not critical to SKIA.

* the generated instuctions are not fully IEEE 754 compilant any more.
* in GCC defintions our IEEE 754 compilance moves one step down: from "Regular" to "Relaxed".

Change-Id: I2416db135545bb24ab8f32b695ecde109e9e35d3
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
---
 Android.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Android.mk b/Android.mk
index b454e54..c68475f 100644
--- a/Android.mk
+++ b/Android.mk
@@ -53,6 +53,7 @@ LOCAL_ARM_MODE := thumb
 
 LOCAL_CFLAGS += \
 	-fPIC \
+	-funsafe-math-optimizations \
 	-Wno-unused-parameter \
 	-U_FORTIFY_SOURCE \
 	-D_FORTIFY_SOURCE=1 \
-- 
2.10.1

